plugins/microservices/src/json_value.cpp

```diff
@@ -0,0 +1,150 @@
+    {
+        IRODS_MSI_REQUIRE_VALID_POINTER(_json_handle);
+        IRODS_MSI_REQUIRE_VALID_POINTER(_json_pointer);
+        IRODS_MSI_REQUIRE_VALID_POINTER(_json_value);
+        IRODS_MSI_REQUIRE_VALID_POINTER(_rei);
+
+        IRODS_MSI_REQUIRE_TYPE(_json_handle->type, STR_MS_T);
+        IRODS_MSI_REQUIRE_TYPE(_json_pointer->type, STR_MS_T);
+
+        IRODS_MSI_REQUIRE_VALID_POINTER(_json_handle->inOutStruct);
+        IRODS_MSI_REQUIRE_VALID_POINTER(_json_pointer->inOutStruct);
+
+        try {
+            auto* handle = static_cast<char*>(_json_handle->inOutStruct);
+            log_msi::debug("JSON handle (address=[{}], value=[{}])", static_cast<void*>(handle), handle);
+
+            auto* object = is::find(handle);
+            if (!object) { // NOLINT(readability-implicit-bool-conversion)
+                log_msi::error("Invalid argument: JSON handle [{}] is not associated with any object.", handle);
+                return SYS_INVALID_INPUT_PARAM;
+            }
+
+            const auto& json_object = boost::any_cast<json&>(*object);
+
+            const auto* p = static_cast<const char*>(_json_pointer->inOutStruct);
+            // NOLINTNEXTLINE(readability-implicit-bool-conversion)
+            log_msi::debug("JSON pointer address=[{}], value=[{}]", static_cast<const void*>(p), (p ? p : "<nullptr>"));
+
+            const json* json_value{};
+
+            try {
+                json_value = &json_object.at(json::json_pointer{p});
+            }
+            catch (const json::out_of_range&) {
+                fillStrInMsParam(_json_value, "<irods::json::out_of_range>");
+                return 0;
+            }
+
+            if (json_value->is_number()) {
+                const auto v = json_value->get<double>(); // Covers integers and floats.
+                fillStrInMsParam(_json_value, std::to_string(v).c_str());
+            }
+            else if (json_value->is_string()) {
+                fillStrInMsParam(_json_value, json_value->get_ref<const std::string&>().c_str());
+            }
+            else if (json_value->is_boolean()) {
+                const auto v = json_value->get<bool>();
+                fillStrInMsParam(_json_value, (v ? "<irods::json::true>" : "<irods::json::false>"));
+            }
+            else if (json_value->is_array()) {
+                fillStrInMsParam(_json_value, "<irods::json::array>");
+            }
+            else if (json_value->is_object()) {
+                fillStrInMsParam(_json_value, "<irods::json::object>");
+            }
+            else if (json_value->is_null()) {
+                fillStrInMsParam(_json_value, "<irods::json::null>");
+            }
+            else if (json_value->is_binary()) {
+                fillStrInMsParam(_json_value, "<irods::json::binary_array>");
+            }
+            else {
+                fillStrInMsParam(_json_value, "<irods::json::type_not_supported>");
```

Comment:
Is this branch even reachable?

To get here means we're able to parse the JSON structure. That means all types are known.

Reply:
The branch is only reachable if the JSON standard introduces support for new types. For now, this line helps to future the server by alerting developers and admins to changes in the available types.